libs/mep/dc0944/aside/aside.js

```diff
@@ -0,0 +1,244 @@
+    else if (textBlocks.length) combineTextBocks(textBlocks, iconArea, viewports[index], variant);
+  });
+  if (variant === 'popup' || el.classList.contains('frictionless')) addCloseButton(el);
+  return foreground;
+}
+
+function decorateLayout(el) {
+  const elems = el.querySelectorAll(':scope > div');
+  if (elems.length > 1) {
+    decorateBlockBg(el, elems[0]);
+  }
+  const foreground = elems[elems.length - 1];
+  foreground.classList.add('foreground', 'container');
+  if (el.classList.contains('promobar')) return decoratePromobar(el);
+  if (el.classList.contains('split')) decorateMedia(el);
+  const text = foreground.querySelector('h1, h2, h3, h4, h5, h6, p')?.closest('div');
+  text?.classList.add('text');
+  const media = foreground.querySelector(':scope > div:not([class])');
+  if (media && !el.classList.contains('notification')) {
+    media.classList.add('image');
+    const video = media.querySelector('video');
+    if (video) applyHoverPlay(video);
+  }
+  const picture = text?.querySelector('p picture');
+  const iconArea = picture ? (picture.closest('p') || createTag('p', null, picture)) : null;
+  iconArea?.classList.add('icon-area');
+  const foregroundImage = foreground.querySelector(':scope > div:not(.text) img')?.closest('div');
+  const bgImage = el.querySelector(':scope > div:not(.text):not(.foreground) img')?.closest('div');
+  const foregroundMedia = foreground.querySelector(':scope > div:not(.text) video, :scope > div:not(.text) a[href*=".mp4"]')?.closest('div');
+  const bgMedia = el.querySelector(':scope > div:not(.text):not(.foreground) video, :scope > div:not(.text):not(.foreground) a[href*=".mp4"]')?.closest('div');
+  const image = foregroundImage ?? bgImage;
+  const asideMedia = foregroundMedia ?? bgMedia ?? image;
+  const isSplit = el.classList.contains('split');
+  const hasMedia = foregroundImage ?? foregroundMedia ?? (isSplit && asideMedia);
+  if (!hasMedia) el.classList.add('no-media');
+  if (asideMedia && !asideMedia.classList.contains('text')) {
+    asideMedia.classList.add(`${isSplit ? 'split-' : ''}image`);
+    if (isSplit) {
+      const position = [...asideMedia.parentNode.children].indexOf(asideMedia);
+      el.classList.add(`split${!position ? '-right' : '-left'}`);
+      foreground.parentElement.appendChild(asideMedia);
+    }
+  } else if (!iconArea) {
+    foreground?.classList.add('no-image');
+  }
+  if (el.classList.contains('split')
+      && (el.classList.contains('medium') || el.classList.contains('large'))) {
+    decorateIconStack(el);
+  }
+  return foreground;
+}
+
+export default function init(el) {
+  el.classList.add('con-block');
+  const blockData = getBlockData(el);
+  const blockText = decorateLayout(el);
+  decorateBlockText(blockText, blockData);
+  decorateStaticLinks(el);
+  formatPromoButton(el);
+  if (el.classList.contains('frictionless')) {
+    const moduleCtas = document.querySelectorAll('.CTAButtons');
+    if (moduleCtas.length) {
```

Comment:
Nit: Same here, we can skip the caching
```suggestion
    if (document.querySelectorAll('.CTAButtons').length) {
```